test_rclcpp/test/test_intra_process.cpp

```diff
@@ -68,8 +72,8 @@ TEST(test_intra_process_within_one_node, nominal_usage) {
     // wait for the first callback
     {
       size_t i = 0;
-      while (counter == 0 && i < 10) {
-        printf("spin_node_once() - callback (1) expected - try %zu/10\n", i++);
+      while (counter == 0 && i < 2) {
```

Comment:
Because I do not think it should take more than that. This condition may not hold in the future, but 2 seemed just as arbitrary as 10 but is stricter.